test/Microsoft.AspNetCore.Mvc.IntegrationTests/ArrayModelBinderIntegrationTest.cs

```diff
@@ -327,5 +327,45 @@ private class Person
             Assert.Equal(0, modelState.ErrorCount);
             Assert.True(modelState.IsValid);
         }
+
+        private class PersonWithReadOnlyAndInitializedProperty
```

Comment:
Coverage looks much better. Don't think we yet have tests covering a `KeyValuePair` or `Dictionary` with a `TValue` that's a `SubModel` containing a `[FromHeader]` property. Believe that's the last remaining gap.